server/src/main/java/org/apache/iotdb/db/mpp/plan/scheduler/FragInstanceDispatchResult.java

```diff
@@ -19,14 +19,27 @@
 
 package org.apache.iotdb.db.mpp.plan.scheduler;
 
+import org.apache.iotdb.common.rpc.thrift.TSStatus;
+
 public class FragInstanceDispatchResult {
   private boolean successful;
 
+  private TSStatus failureStatus;
```

Comment:
Maybe we can add one more parameter such as `message` and construct it when dispatch failed.

The error message can be fetched from remote response or local dispatch result.

And furthermore, we can change the `failureStatus` to `status` and remove the original param `successful` because it seems be duplicated.

Reply:
The message can be obtained from the failureStatus.